bin/projects/dbatools/dbatools/Parameter/DbaInstanceParameter.cs

```diff
@@ -279,18 +279,37 @@ public DbaInstanceParameter(string Name)
             // Connection String interpretation
             try
             {
-                System.Data.SqlClient.SqlConnectionStringBuilder connectionString = new System.Data.SqlClient.SqlConnectionStringBuilder(tempString);
+                System.Data.SqlClient.SqlConnectionStringBuilder connectionString =
+                    new System.Data.SqlClient.SqlConnectionStringBuilder(tempString);
                 DbaInstanceParameter tempParam = new DbaInstanceParameter(connectionString.DataSource);
                 _ComputerName = tempParam.ComputerName;
-                if (tempParam.InstanceName != "MSSQLSERVER") { _InstanceName = tempParam.InstanceName; }
-                if (tempParam.Port != 1433) { _Port = tempParam.Port; }
+                if (tempParam.InstanceName != "MSSQLSERVER")
+                {
+                    _InstanceName = tempParam.InstanceName;
+                }
+                if (tempParam.Port != 1433)
+                {
+                    _Port = tempParam.Port;
+                }
                 _NetworkProtocol = tempParam.NetworkProtocol;
 
                 IsConnectionString = true;
 
                 return;
             }
-            catch { }
+            catch (ArgumentException ex)
+            {
+                string name = "unknown";
+                try
+                {
+                    name = ex.TargetSite.GetParameters()[0].Name;
+                }
+                catch { }
+                if (name == "keyword")
+                {
+                    throw;
+                }
+            }
```

Comment:
still missing an empty catch for all the other exceptions that are not argument exceptions.